src/Common/isLocalAddress.cpp

```diff
@@ -34,6 +36,12 @@ struct NetworkInterfaces
         }
     }
 
+    void swap(NetworkInterfaces && other)
+    {
+        ifaddr = other.ifaddr;
```

Comment:
Well, `unique_ptr` was convenient to use since it's called destructor automatically (and less code). W/o it, the destructor should be called here manually :(

IMO, just adding a method like `reset()` would be sufficient (don't see any reasons for `swap()`). But probably it's already nitpicking.

Reply:
`NetworkInterfaces` is non-copyable, so I think we need `swap` here, not just `reset`. But my implementation here is incorrect. It should be:

```
auto tmp = ifaddr;
ifaddr = other.ifaddr;
other.ifaddr = tmp;
```

Also don't need call destructor manually. After swapping, `other` will be destroyed and the old `ifaddr` is freed.